src/cfnlint/maintenance.py

```diff
@@ -58,6 +58,28 @@ def update_resource_spec(region, url):
     with open(filename, 'w') as f:
         json.dump(spec, f, indent=2, sort_keys=True, separators=(',', ': '))
 
+    lines = []
+    botocore_cache = {}
+    with open(filename, 'r') as f:
+        for line in f:
+            if 'botocore' in line:
+                service_and_type = line.split('"')[3]
+                service = '/'.join(service_and_type.split('/')[:-1])
+                botocore_type = service_and_type.split('/')[-1]
+                if service not in botocore_cache:
+                    botocore_cache[service] = json.loads(get_url_content('https://raw.githubusercontent.com/boto/botocore/master/botocore/data/' + service + '/service-2.json'))
```

Comment:
Another option here is
```
botocore_cache[service] = json.loads(pkgutil.get_data('botocore', os.path.join('data', service_and_type[0], service_and_type[1], 'service-2.json')))
```

While we don't directly rely on botocore it is getting loaded because of the SAM dependencies.

Reply:
This is the biggest decision I've been debating
This would speed things up significantly, but would grab outdated information if `botocore` is outdated

```bash
$ time cfn-lint --update-specs # pulling from botocore latest on Github 
real  2m15.260s

$ time pip3 install botocore --upgrade
real	0m15.649s
$ time cfn-lint --update-specs # pulling directly from local copy of botocore
real	0m24.975s


# might be able to pull latest data faster than individual requests

$ time git clone https://github.com/boto/botocore/ --depth 1
real	0m7.271s

$ time curl -s -L https://codeload.github.com/boto/botocore/zip/master > botocore.zip
real	0m2.605s
$ time unzip botocore.zip
real	0m2.074s
```